### PR TITLE
feat: agent install version & version checking on connection & minor team selector fix

### PR DESCRIPTION
**Pull request by robot9706**
- When changing teams the browser does not correctly reload showing the old team
- Crux uses an environment variable for agent installation, replaced with major + minor from package version
- When an agent connects the version is matched against crux to make sure it is compatible (only in PRODUCTION mode)

### CODE REVIEW

**Comment by robot9706** [comment]
Should I use 'latest' in non production mode so development is easier or everybody should just set `CRUX_AGENT_IMAGE` in their .envs?

**Review by polaroi8d — CHANGES_REQUESTED** [review]
I'm missing the unit tests for `getAgentVersionFromPackage` and `agentVersionSupported`.

**Review by polaroi8d — APPROVED** [review]

File: web/crux/src/shared/package.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { coerce, major, minor } from 'semver'
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const pkg = require('../../package.json')
```
Comment (m8vago) [inline web/crux/src/shared/package.ts:4]: Use `const version = this.configService.get<string>('npm_package_version')`

File: web/crux/src/app/agent/agent.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        publicKey: 'key',
+      }
+
+      const agentObs = agentService.handleConnect(new GrpcNodeConnectionMock(), info)
+      const agentSub = agentObs.subscribe(() => {})
+
+      expect(agentSub.closed).toBe(false)
+    })
+
+    it('handleConnect should close connection if an agent has a non supported version', async () => {
```
[the remark this answers is not in the document]
Comment (polaroi8d) [reply web/crux/src/app/agent/agent.service.spec.ts:106]: Hmm good case, I think If you update the instance and the agent tries to connect with the wrong version, we should easily just kick them. (After that you have to update agents as well, to workaround your problem)

[the remark this answers is not in the document]
Comment (m8vago) [reply web/crux/src/app/agent/agent.service.spec.ts:106]: But we have the node update feature and if we kick them immediately the users can not update their agent without revoking the token.

[the remark this answers is not in the document]
Comment (polaroi8d) [reply web/crux/src/app/agent/agent.service.spec.ts:106]: Then what, If we automatically update the agents if the platform version is bumped?

[the remark this answers is not in the document]
Comment (m8vago) [reply web/crux/src/app/agent/agent.service.spec.ts:106]: Conclusion: We will try to update once.
If it failes we shut down the agent. 
The ultimate solution will be developed in a different PR.

**Review by m8vago — APPROVED** [review]

**Pull request merged**